Implement Search ListItems and table header

luacmartins opened the pull request:
### Details
Introduces the TransactionListItem and SearchTableHeader to show list items in the search page

### Fixed Issues
$ https://github.com/Expensify/App/issues/39876
$ https://github.com/Expensify/App/issues/39877

### Tests
1. Navigate to `Profile > Troubleshoot > New Search Page`
2. Verify that you can see the table header and the transaction items.
3. Verify that clicking on a transaction item opens the transaction thread in the RHP

- [x] Verify that no errors appear in the JS console

### Offline tests

### QA Steps
Same as tests

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I verified that similar component doesn't exist in the codebase
- [x] I verified that all props are defined accurately and each prop has a `/** comment above it */`
- [x] I verified that each file is named correctly
- [x] I verified that each component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
- [x] I verified that the only data being stored in component state is data necessary for rendering and nothing else
- [x] In component if we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
- [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
- [x] I verified that component internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
- [x] I verified that all JSX used for rendering exists in the render method
- [x] I verified that each component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android: mWeb Chrome</summary>

![chrome](https://github.com/Expensify/App/assets/22219519/2f2724e0-0708-44ad-b769-dc800bce810a)

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: Native</summary>



<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: mWeb Safari</summary>


<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

![web](https://github.com/Expensify/App/assets/22219519/37229c0e-d6cb-442e-b418-833c9b6cce94)


<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Desktop</summary>

<img width="1198" alt="desktop" src="https://github.com/Expensify/App/assets/22219519/c12a8c0d-4270-4e78-a5ee-f167727bbbcc">

<!-- add screenshots or videos here -->

</details>

luacmartins committed:
create searchtableheader

luacmartins committed:
fix header style

luacmartins committed:
create receipt element

luacmartins committed:
update row callback

luacmartins committed:
add button translation and styles

luacmartins committed:
use display amounts

luacmartins committed:
refactor tag element

luacmartins committed:
create getSearchTableColumnWidth

luacmartins committed:
add styles framework

luacmartins committed:
dry user element

luacmartins committed:
style description

luacmartins committed:
dry code

luacmartins committed:
dry more code

luacmartins committed:
rm unused code

luacmartins committed:
use modified values if present

luacmartins committed:
rm getCleanResults

luacmartins committed:
inject from/to into listitem

luacmartins committed:
refactor code

luacmartins committed:
style narrow view

luacmartins committed:
style narrow view

luacmartins committed:
refactor cells

luacmartins committed:
fix lint

luacmartins committed:
fix lint, typecheck

luacmartins committed:
fix typecheck

luacmartins committed:
fix lint

luacmartins committed:
fix amounts, use utils methods

luacmartins committed:
fix avatars

luacmartins committed:
fix ts

luacmartins committed:
do not show default merchant

luacmartins committed:
update logic to show merchant or description

luacmartins committed:
rm debug log

luacmartins committed:
fix lint

luacmartins committed:
add keyForList

luacmartins committed:
fix ts

luacmartins committed:
fix lint

luacmartins commented:
A couple of known issues:

- Opening card transactions leads to a not found page. This happens because `transactionThreadReportID` doesn't exist for some of those. 
- Resizing the screen to switch from the large screen to small screen breaks the back navigation on RHP.

luacmartins commented:
Running adhoc build [here](https://github.com/Expensify/App/actions/runs/8915600314)

luacmartins committed:
fix styles on native

luacmartins committed:
fix lint

shawnborton commented:
Is the view button here just as a placeholder for the action column? I would think that View is a button we don't need to show since tapping on the row will view the expense. Instead, we'd just save this column for things like Approve, Pay, etc.

shawnborton commented:
Dang, this is feeling pretty good! Part of me wonders if we should actually use a bigger font in the desktop table since it seems like we actually have a decent bit of room? Right now on desktop, most of the table data uses label font size (13px) but I wonder if we should bump that to our standard font size (15px)? Maybe we can use a medium breakpoint for when we are in between mobile and wider desktops and use label for that instead.

shawnborton commented:
cc @Expensify/design for thoughts on that ^

shawnborton commented:
Can we check the paddings here? It looks like the top button is wider than the rows below. I think everything should sit within 20px of horizontal padding:
![CleanShot 2024-05-01 at 21 05 34@2x](https://github.com/Expensify/App/assets/2319350/d375ff20-a73d-456c-afda-dd38481b0a30)

dubielzyk-expensify commented:
> Dang, this is feeling pretty good! Part of me wonders if we should actually use a bigger font in the desktop table since it seems like we actually have a decent bit of room? Right now on desktop, most of the table data uses label font size (13px) but I wonder if we should bump that to our standard font size (15px)? Maybe we can use a medium breakpoint for when we are in between mobile and wider desktops and use label for that instead.

I'm okay with this. There's a few differences on mobile vs desktop anyways, so I don't think this need the level of consistency as elsewhere.

luacmartins committed:
fix merchant logic

luacmartins committed:
fix display name overflow

luacmartins committed:
fix button padding

luacmartins commented:
Collecting all the bugs here:

1. [The list items flickers when scrolling the list quickly or arrive at the end of the list](https://github.com/Expensify/App/pull/41347#issuecomment-2089286237)

I've noticed this one too, I'll say this is NAB for now. I'll have to investigate this, but maybe will open it to contributors.

2. [The description column is shown in search page instead of the merchant column](https://github.com/Expensify/App/pull/41347#issuecomment-2089301176)

Fix in my [latest commit](https://github.com/Expensify/App/pull/41347/commits/de81594f647da58c877ef48d012ef1d4fe8a8951)

3. [The modification is not applied to the list item, you need to refresh the page for the modification to appear](https://github.com/Expensify/App/pull/41347#issuecomment-2089303007)

This feature is not available yet. It will come in phase 2 of the project

4. [Long display name / email is not truncated in list item](https://github.com/Expensify/App/pull/41347#issuecomment-2089330517)

Fixed in [this commit](https://github.com/Expensify/App/pull/41347/commits/07bb6418fa4b05b9afdba368c4e584d14dd3b89e)

5. [When scrolling up, the list jumps down](https://github.com/Expensify/App/pull/41347#issuecomment-2089355871)

I've noticed this too. I'd say this is NAB and we'd address in a follow up.


5. Is the view button here just as a placeholder for the action column? I would think that View is a button we don't need to show since tapping on the row will view the expense. Instead, we'd just save this column for things like Approve, Pay, etc.

6. [Button padding](https://github.com/Expensify/App/pull/41347#issuecomment-2089363059)

Fixed in [this commit](https://github.com/Expensify/App/pull/41347/commits/101439499ac122cd0e4eb62f3ed033b6d8d55388)

luacmartins commented:
>Is the view button here just as a placeholder for the action column? I would think that View is a button we don't need to show since tapping on the row will view the expense. Instead, we'd just save this column for things like Approve, Pay, etc.

@shawnborton Kind of. It seems like once we introduce the other action buttons, we'd show the `View` button for consistency. Right now the only available option is `View` since the other options won't come until phase 2. Would you rather remove them for now?

luacmartins committed:
fix lint

luacmartins commented:
So there are the remaining bugs:

- Opening card transactions leads to a not found page. This happens because transactionThreadReportID doesn't exist for some of those.
- Resizing the screen to switch from the large screen to small screen breaks the back navigation on RHP.
- [The list items flickers when scrolling the list quickly or arrive at the end of the list](https://github.com/Expensify/App/pull/41347#issuecomment-2089286237)
- [When scrolling up, the list jumps down](https://github.com/Expensify/App/pull/41347#issuecomment-2089355871)

grgia commented:
Rerunning the failed builds ^

grgia commented on src/components/Search.tsx:
```diff
@@ -16,35 +13,16 @@ import ROUTES from '@src/ROUTES';
 import TableListItemSkeleton from './Skeletons/TableListItemSkeleton';
-import Text from './Text';
-
-/**
- * Todo This is a temporary function that will pick search results from under `snapshot_` key
- * either api needs to be updated to key by `snapshot_hash` or app code calling search data needs to be refactored
- * remove this function once this is properly fixed
- */
```
Which PR fixed this?

luacmartins replied:
This PR

grgia commented on src/components/SelectionList/SearchTableHeader.tsx:
```diff
@@ -0,0 +1,57 @@
+import CONST from '@src/CONST';
+
+type SearchTableHeaderProps = {
+    /** Whether we should show the merchant or description column */
+    shouldShowMerchant: boolean;
```
For the future cases of mixed reports/expenses- would there be a case where we'd show both the merchant and the description?

luacmartins replied:
It's possible, but that'd be a future consideration

grgia commented on src/components/SelectionList/SearchTableHeader.tsx:
```diff
@@ -0,0 +1,57 @@
+    const {translate} = useLocalize();
+    const displayNarrowVersion = isMediumScreenWidth || isSmallScreenWidth;
+
+    if (displayNarrowVersion) {
+        return;
+    }
```
isSmallScreen is the typical variable name we'd use to indicate mobile size

luacmartins replied:
Well, this includes both small and medium size screens, what name do we use for that?

luacmartins replied:
Gonna use isLargeScreenWidth instead

luacmartins replied:
Done

grgia commented on src/components/SelectionList/SearchTableHeader.tsx:
```diff
@@ -0,0 +1,57 @@
+                <View style={[StyleUtils.getSearchTableColumnStyles(CONST.SEARCH_TABLE_COLUMNS.DATE)]}>
+                    <Text style={[styles.mutedNormalTextLabel]}>{translate('common.date')}</Text>
+                </View>
+                <View style={[StyleUtils.getSearchTableColumnStyles(CONST.SEARCH_TABLE_COLUMNS.MERCHANT)]}>
+                    <Text style={[styles.mutedNormalTextLabel]}>{translate(shouldShowMerchant ? 'common.merchant' : 'common.description')}</Text>
+                </View>
```
Could you explain why merchant/description are grouped

luacmartins replied:
Right now we're showing one or the other, following this logic in the HL

>The Merchant and Description columns will show based on the same logic already in use for the expense previews in Inbox. In other words, Merchant will always be shown for workspace expenses, because Merchant is required and triggers violations. Off-workspace, we’ll show Merchant as long as at least one value isn’t “Request” or “Unknown merchant”, based on existing logic. In all other cases, we’ll show Description as long as at least one expense has one (and if a description is long, we’ll truncate with an ellipsis, just like any other column).

grgia commented:
<img width="1449" alt="Screenshot 2024-05-02 at 1 30 56 PM" src="https://github.com/Expensify/App/assets/38015950/0ec397f7-2f7f-40c4-9439-4bd666b280a8">


<img width="958" alt="image" src="https://github.com/Expensify/App/assets/38015950/444cdbcc-b4e0-4e23-901d-75860915c693">


- The search query is returning reports that are not mine on my @expensify account @luacmartins - like that 20K industrious one... actually, none of them are mine 😅

- The workspace avatars are not correctly colored (might be missing a prop here)

- We shouldn't have __FAKE__ reports included in the query if we're paginating 

- If we're going to make the rows clickable, shouldn't we add the arrows on the right? Shouldn't these navigate to the page in the RHN?

grgia commented:
<img width="752" alt="image" src="https://github.com/Expensify/App/assets/38015950/e63ffb03-2e86-44dd-bad4-2f6606b7f5e4">

I believe this height should be 64 total not 62

grgia commented on src/styles/index.ts:
```diff
@@ -2665,7 +2665,7 @@ const styles = (theme: ThemeColors) =>
             fontSize: variables.fontSizeMedium,
             color: theme.text,
         },
-        signInButtonAvatar: {
```
typically we would do something like `StyleUtils.getWidthStyle(variables.SOME_CONST)` instead

luacmartins replied:
Oh nice! I'll work on that

grgia commented on src/styles/utils/index.ts:
```diff
@@ -1522,6 +1522,37 @@ const createStyleUtils = (theme: ThemeColors, styles: ThemeStyles) => ({
+        let columnWidth;
+        switch (columnName) {
+            case CONST.SEARCH_TABLE_COLUMNS.DATE:
+                columnWidth = {width: 44};
```
all of these CONST widths should be added to `variables.ts`

luacmartins replied:
Done

shawnborton commented:
> I believe this height should be 64 total not 62

I guess right now, without the receipt thumbnails, we should get 12px top padding + 28px tall row to account for the button height + 12px bottom padding, **which is 52px total**. Once we get the receipt thumbnail in there, we'd have 12px top padding + 40px tall receipt + 12px bottom padding, which is 64px. 

> If we're going to make the rows clickable, shouldn't we add the arrows on the right? 

We got rid of the right arrows in our designs to save some valuable real estate.

grgia commented:
All bar should have no caret/not be clickable if there are no further options

<img width="818" alt="image" src="https://github.com/Expensify/App/assets/38015950/639d59fb-e174-4b54-b636-e44a421d61ae">

grgia commented:
@Expensify/design could I get a link to the final mocks for search v1?

shawnborton commented:
Figma is [here](https://www.figma.com/file/YHaWaw3F0PgxqI0tXHJVeY/Search-v1-v3?type=design&node-id=13%3A21337&mode=design&t=IWKKzWSL23B7jPR5-1), I think you are already in there! 🙈

grgia commented:
@luacmartins I think this fixes the incorrect transactions - https://github.com/Expensify/Auth/pull/10764

luacmartins committed:
use variables

luacmartins committed:
update row vertical padding

luacmartins committed:
rm button padding

luacmartins committed:
use isLargeScreenWidth

luacmartins commented:
>I guess right now, without the receipt thumbnails, we should get 12px top padding + 28px tall row to account for the button height + 12px bottom padding, which is 52px total.

Done

luacmartins committed:
fix avatar colors

luacmartins committed:
use style utils

luacmartins committed:
rm receipt styles

luacmartins committed:
fix left aligned items

luacmartins committed:
rename to expenses

luacmartins commented on src/components/SelectionList/TransactionListItem.tsx:
```diff
@@ -40,120 +41,105 @@ function TransactionListItem<TItem extends ListItem>({
-        />
-    );
+    const userCell = (participant: SearchPersonalDetails & SearchPolicyDetails) => {
+        const displayName = participant.name ?? participant.displayName ?? participant.login;
```
What does your data look like for that participant?

luacmartins replied:
I haven't been able to reproduce that yet. We had some API PRs deployed earlier today, could you try cleaning your data and trying again?

luacmartins commented:
@shawnborton this is what using 15px for date/merchant and 13px for from/to looks like:
![Screenshot 2024-05-02 at 1 32 02 PM](https://github.com/Expensify/App/assets/22219519/91b78b77-25da-4736-857d-bee42809265e)

shawnborton commented:
> this is what using 15px for date/merchant and 13px for from/to looks like:

Okay cool, I don't mind this but I think we need to wait until there are more columns (checkbox, receipt, etc) before fully making a decision here.

luacmartins commented:
@shawnborton ok, should I update the font to 15/13px now or keep the current 13/11px for now?

shawnborton commented:
Yeah that's my thinking too, let's stay the course with 13/11 for now

luacmartins committed:
use mw50

luacmartins commented:
>[All bar should have no caret/not be clickable if there are no further options](https://github.com/Expensify/App/pull/41347#issuecomment-2090413242)
For this one, would we just remove the button completely? It doesn't make much sense to keep a disabled button with no functionality in App

luacmartins committed:
fix crash

shawnborton commented:
> For this one, would we just remove the button completely? It doesn't make much sense to keep a disabled button with no functionality in App

On mobile, yup - I think that makes the most sense for now.

luacmartins commented:
@shawnborton this is what it'd look like then:
 
![Screenshot 2024-05-02 at 1 54 21 PM](https://github.com/Expensify/App/assets/22219519/ea536bce-3e5c-497c-b422-a76f0a77bd79)

shawnborton commented:
Cool, that works for me!

luacmartins committed:
rm button on mobile

luacmartins committed:
fix lint

luacmartins commented:
Cool, so this is a list of comments addressed so far:

- [x] [Row height should be 52px](https://github.com/Expensify/App/pull/41347#issuecomment-2090407253)
- [x] [Rename displayNarrowVersion to isSmallScreen](https://github.com/Expensify/App/pull/41347#discussion_r1587535414)
- [x] [The workspace avatars are not correctly colored (might be missing a prop here)](https://github.com/Expensify/App/pull/41347#issuecomment-2090398222)
- [x] [The -> and the TO User should be left aligned](https://github.com/Expensify/App/pull/41347#issuecomment-2090413242)
- [x] [Also, I spoke with @trjExpensify and we might just want to rename "All" to "Expenses" for now until we have more options.](https://github.com/Expensify/App/pull/41347#issuecomment-2090445914)
- [x] [Increase font size on large screens](https://github.com/Expensify/App/pull/41347#issuecomment-2091423629)
- [x] [All bar should have no caret/not be clickable if there are no further options](https://github.com/Expensify/App/pull/41347#issuecomment-2090413242)
- [x] [App crash](https://github.com/Expensify/App/pull/41347#issuecomment-2089905365)
- [x] [Incorrect transactions](https://github.com/Expensify/App/pull/41347#issuecomment-2090563018)
- [x] [If we're going to make the rows clickable, shouldn't we add the arrows on the right? Shouldn't these navigate to the page in the RHN?](https://github.com/Expensify/App/pull/41347#issuecomment-2090407253)
- [x] [Use getWidthStyle](https://github.com/Expensify/App/pull/41347#discussion_r1587572473)
- [x] [Add const widths to variables.ts](https://github.com/Expensify/App/pull/41347#discussion_r1587572473)

This are the ones still pending:

- [ ] Opening card transactions leads to a not found page. This happens because transactionThreadReportID doesn't exist for some of those.
- [ ] We shouldn't have FAKE reports included in the query if we're paginating
- [ ] Resizing the screen to switch from the large screen to small screen breaks the back navigation on RHP.
- [ ] [The list items flickers when scrolling the list quickly or arrive at the end of the list](https://github.com/Expensify/App/pull/41347#issuecomment-2089286237)
- [ ] [When scrolling up, the list jumps down](https://github.com/Expensify/App/pull/41347#issuecomment-2089355871)

The first 2 involve API changes, so NAB for this PR. The remaining issues will be tackled as a follow up since they are quite involved and we wanna get this functionality out.

luacmartins commented:
@grgia @rayane-djouah could you please re-review?

luacmartins commented:
Kicked off another adhoc build [here](https://github.com/Expensify/App/actions/runs/8929829758)

shawnborton commented:
Can we bump this gap up to 8px? The space between the names and the action button: 
![CleanShot 2024-05-02 at 16 33 01@2x](https://github.com/Expensify/App/assets/2319350/ae6f9b26-9dfd-4423-9d7a-cc18ae7e7220)

luacmartins committed:
increase gap

luacmartins commented:
@shawnborton done

luacmartins commented:
@rayane-djouah are you able to continue your review on this one please?

rayane-d reviewed (approved):
Changes look good and test well. The reviewer checklist is also complete.

shawnborton commented:
Not related to this PR but seeing some weirdness with some of the expense views in the RHP:
![CleanShot 2024-05-02 at 20 33 17@2x](https://github.com/Expensify/App/assets/2319350/e6711623-2ebc-4db6-8df1-ea7ee67d36e2)

As in, I shouldn't see a Join button and the header styles are a bit different from a proper report/expense in the RHP:
![CleanShot 2024-05-02 at 20 34 14@2x](https://github.com/Expensify/App/assets/2319350/54bdd094-254f-4acc-ad68-83ef1fb965fd)

grgia commented:
@rayane-djouah are you able to build IOS without error

grgia reviewed (approved):
changes LGTM, have we created issues for all the follow ups?

grgia commented:
Coming from [here](https://expensify.slack.com/archives/C036QM0SLJK/p1714723383109589?thread_ts=1714723069.945699&cid=C036QM0SLJK) should we sort by date descending to begin with? The transactions feel randomly organized right now

grgia commented on src/styles/utils/index.ts:
```diff
@@ -1522,6 +1522,37 @@ const createStyleUtils = (theme: ThemeColors, styles: ThemeStyles) => ({
+        let columnWidth;
+        switch (columnName) {
+            case CONST.SEARCH_TABLE_COLUMNS.DATE:
+                columnWidth = getWidthStyle(variables.w44);
```
NAB these variables are typically named by their use, so like `variables.dateColumnWidth`

grgia commented:
<img width="644" alt="image" src="https://github.com/Expensify/App/assets/38015950/c6008919-5af6-47e0-bcb2-e611db3e9349">
This should still be fixed, let me take a look

luacmartins commented:
Gonna merge this one to get the feature out and then create a follow up to address the remaining issues. Thanks for the reviews everyone!

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

luacmartins commented:
Created issues here:

- https://github.com/Expensify/App/issues/41578
- https://github.com/Expensify/App/issues/41579
- https://github.com/Expensify/App/issues/41581
- https://github.com/Expensify/App/issues/41582
- https://github.com/Expensify/App/issues/41585

shawnborton commented:
Some more follow ups - the headers don't seem aligned to the content below:
![CleanShot 2024-05-03 at 14 53 45@2x](https://github.com/Expensify/App/assets/2319350/ab5b0a54-180c-4213-a832-4be4a0980794)

cc @Expensify/design

luacmartins commented:
@shawnborton I don't see that, which version are you using?

shawnborton commented:
Yup I am using that adhoc build and noticed this.

luacmartins commented:
Weird, I'm on the same build and don't see that
![Screenshot 2024-05-03 at 1 21 30 PM](https://github.com/Expensify/App/assets/22219519/12eac026-fd5d-4f9e-a2c8-089a548a8b59)

luacmartins commented:
I do see some issues on iOS with the from/to fields though
![simulator_screenshot_1BAE352D-1A58-4D44-B471-E36B7E7EF764](https://github.com/Expensify/App/assets/22219519/3b8a5b7d-08ac-4f71-a3bf-546cf5440d37)

shawnborton commented:
It's probably because I have a mouse plugged in and the fixed header is above the scrollable area? And then the scroll bar pushes everything over? 
![CleanShot 2024-05-03 at 15 29 50@2x](https://github.com/Expensify/App/assets/2319350/14a9fb70-0e22-4d5d-9bd6-3288a22076f8)

luacmartins commented:
Hmm I didn't test it with a mouse but I also just see the scrollbar below the header

https://github.com/Expensify/App/assets/22219519/76c0a5ef-2397-44fa-beda-40863c443f51

luacmartins commented:
@shawnborton do you see an additional right padding on the header? I see this padding, which makes the alignment correct
![Screenshot 2024-05-03 at 2 41 46 PM](https://github.com/Expensify/App/assets/22219519/4d5b1e04-7cf3-4714-abed-3a1c2bbd15ed)

shawnborton commented:
Try testing it with a mouse though - I think the mouse creates that fixed scrollbar that is pushing the table to be over to the left a bit.

shawnborton commented:
Sounds good!

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9011216944) to production by https://github.com/marcaaron in version: 1.4.71-6 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅
